lib/vtls/mbedtls.c

```diff
@@ -1069,7 +1114,11 @@ static CURLcode mbedtls_sha256sum(const unsigned char *input,
   mbedtls_sha256(input, inputlen, sha256sum, 0);
 #else
   /* returns 0 on success, otherwise failure */
+#if MBEDTLS_VERSION_MAJOR >= 3
+  if(mbedtls_sha256(input, inputlen, sha256sum, 0) != 0)
+#else
   if(mbedtls_sha256_ret(input, inputlen, sha256sum, 0) != 0)
+#endif
```

Comment:
The fact that we now use the same function name in 3 that we used in 2.0.x which was deprecated in favour of `_ret` in 2.x needs a pretty big comment explaining how that mess is strung together.

Also, while not a fault of this PR, blimey thats ugly..